src/utils/modal-logic/constrained-modal-logic-connectives.jl

```diff
@@ -0,0 +1,204 @@
+"""
+struct ConstrainedConnective{S,N,F} <: Connective
+    condition::F
+
+    ConstrainedConnective{S,N}(condition::F) where {S,N,F<:Function} = new{S,N,F}(condition)
+
+    ConstrainedConnective{S}(
+        threshold::Int,
+        condition::F
+    ) where {S,F<:Function} = new{S,threshold,F}(condition)
+
+end
+
+"""
+    name(::ConstrainedConnective{S}) where {S} = S
+
+Return the symbol encapsulated by a [`ConstrainedConnective`](@ref).
+
+See also [`Connective`](@ref).
+"""
+name(::ConstrainedConnective{S,N}) where {S,N} = S
+
+"""
+    condition(cc::ConstrainedConnective) = cc.condition
+
+Return a comparator wrapped within a [`ConstrainedConnective`](@ref).
+It is a special function intended to compare the set of neighbors of a specific world within
+ an [`AbstractFrame`](@ref), with respect to an integer `threshold`.
+
+See also [`Connective`](@ref), [`DIAMOND`](@ref), [`DIAMOND2`](@ref), [`threshold`](@ref).
+"""
+condition(cc::ConstrainedConnective) = cc.condition
+
+"""
+    condition(cc::ConstrainedConnective, val::Int)
+
+Shortcut for `condition(cc)(val, threshold(cc))`.
+
+See also [`condition(cc::ConstrainedConnective)`](@ref), [`ConstrainedConnective`](@ref).
+"""
+condition(cc::ConstrainedConnective, val::Int) = condition(cc)(val, threshold(cc))
+
+
+"""
+    threshold(cc::ConstrainedConnective) = cc.threshold
+
+Local argument of [`condition(cc::ConstrainedConnective)`](@ref).
+"""
+threshold(::ConstrainedConnective{S,N}) where {S,N} = N
+
+syntaxstring(cc::ConstrainedConnective; kwargs...) = (name(cc), threshold(cc)) |> join
+Base.show(io::IO, cc::ConstrainedConnective) = print(io, "$(syntaxstring(cc))")
+
+
+"""
+    const DIAMOND2 = NamedConnective{:◊}(>=, 2)
+    const ◊₂ = DIAMOND2
+
+Special [`DIAMOND`](@ref) connective, stating that there are at least 2 accessible worlds
+(within an [`AbstractFrame`](@ref) where a formula holds.
+
+See [`Connective`](@ref), [`DIAMOND`](@ref), [`ConstrainedConnective`](@ref).
+"""
+const DIAMOND2 = ConstrainedConnective{:◊,2}(>=)
+const ◊₂ = DIAMOND2
+
+"""
+    const DIAMOND3 = ConstrainedConnective{:◊}(>=, 3)
+    const ◊₃ = DIAMOND3
+
+See [`DIAMOND2`](@ref), [`ConstrainedConnective`](@ref).
+"""
+const DIAMOND3 = ConstrainedConnective{:◊,3}(>=)
+const ◊₃ = DIAMOND3
+
+"""
+    const ◊ₙ = DIAMOND2
+
+This is just a placeholder for [`DIAMOND2`](@ref).
+Semantically, you can use this to represent a generic [`ConstrainedConnective`](@ref) wrapping
+the lozenge glyph.
+
+When defining the traits for a `ConstrainedConnective{:◊}`, everything is forwarded from the
+traits of `NamedConnective{:◊}` (whose placeholder is just `const ◊`, or [`DIAMOND`](@ref)).
+
+The [`dual`](@ref) connective of `DIAMOND2` is [`BOX2`](@ref), as (¬□₂¬φ) translates to
+"it is not true that there are fewer than 2 accessible worlds where φ holds".
+
+See also [`ismodal`](@ref), [`isdiamond`](@ref), [`isbox`](@ref), [`arity`](@ref),
+[`precedence`](@ref), [`associativity`](@ref), [`□ₙ`](@ref).
+"""
+const ◊ₙ = ◊₂
```

Comment:
If we make this a constant, then user won't be able to use this symbol for their use case. What if we provide a generic constructor instead
```suggestion
◊ₙ(n::Int64) = GradedConnective{...,n}(>=)
```
?

Then one can for example define a logic with, say `◊ₙ.(1:10)`